### PR TITLE
[Bugfix] Renames in 0.15.0 diffusers

### PR DESCRIPTION
**Pull request by StAlKeR7779**
Link to PR in diffusers repository:
https://github.com/huggingface/diffusers/pull/2691

Imports:
`diffusers.models.cross_attention -> diffusers.models.attention_processor`

Unions:
`AttnProcessor -> AttentionProcessor`

Classes:
| Old name | New name |
| --- | --- |
| CrossAttention | Attention |
| CrossAttnProcessor | AttnProcessor |
| XFormersCrossAttnProcessor | XFormersAttnProcessor |
| CrossAttnAddedKVProcessor | AttnAddedKVProcessor |
| LoRACrossAttnProcessor | LoRAAttnProcessor |
| LoRAXFormersCrossAttnProcessor | LoRAXFormersAttnProcessor |
| FlaxCrossAttention | FlaxAttention |
| AttendExciteCrossAttnProcessor | AttendExciteAttnProcessor |
| Pix2PixZeroCrossAttnProcessor | Pix2PixZeroAttnProcessor |

Also config values no longer sets as attributes of object:
https://github.com/huggingface/diffusers/pull/2849

### CODE REVIEW

File: ldm/invoke/generator/diffusers_pipeline.py [hunk 1]
```diff
@@ -400,7 +400,7 @@ def device(self) -> torch.device:
     @property
     def _submodels(self) -> Sequence[torch.nn.Module]:
         module_names, _, _ = self.extract_init_dict(dict(self.config))
-        values = [getattr(self, name) for name in module_names.keys()]
+        values = [getattr(self, name, getattr(self.config, name)) for name in module_names.keys()]
```
Comment (StAlKeR7779) [inline ldm/invoke/generator/diffusers_pipeline.py:403]: I think it's correctly to do:
`getattr(self, name) if hasattr(self, name) else getattr(self.config, name)`
But... maybe it's better to do not in short form then?
```python
submodels = []
for name in module_names.keys():
    if hasattr(self. name):
        value = getattr(self, name)
    else:
        value = getattr(self.config, name)
    if isinstance(value, torch.nn.Module):
        submodels.append(value)
return submodels
```
Maybe this method can be simplified, but I not sure - can modules be in config or only in self object or in both.

Comment (GreggHelt2) [reply ldm/invoke/generator/diffusers_pipeline.py:403]: Would this work with just dropping the attempt to getattr(self, name) and only use getattr(self.config, name)?

`values = [getattr(self.config, name)) for name in module_names.keys()]`

Based on  the [comments on the diffusers PR 2849](https://github.com/huggingface/diffusers/pull/2849#discussion_r1150529864) you mentioned, it sounds like they are trying to eliminate the possible redundancy of having to look for these values in both self attr and self.config attr, and move to only keeping in self.config.
mentioned:

Comment (StAlKeR7779) [reply ldm/invoke/generator/diffusers_pipeline.py:403]: Test this - if take values only from configs, there other errors appears.
So - we need get value from both self and self.config.

**Comment by StAlKeR7779** [comment]
@lstein, @blessedcoolant Looks like i'm done with it.

**Comment by lstein** [comment]
Great and thanks! As I noted in Discord, I'm going to do a hotfix
2.3.4.post1 release with pinned 0.14 today, and then a 2.3.5 release with
your PR merged to allow time for a pre-release and  end-user testing.

Lincoln

File: ldm/invoke/generator/diffusers_pipeline.py [hunk 2]
```diff
@@ -400,8 +400,15 @@ def device(self) -> torch.device:
     @property
     def _submodels(self) -> Sequence[torch.nn.Module]:
         module_names, _, _ = self.extract_init_dict(dict(self.config))
-        values = [getattr(self, name) for name in module_names.keys()]
-        return [m for m in values if isinstance(m, torch.nn.Module)]
+        submodels = []
+        for name in module_names.keys():
+            if hasattr(self, name):
```
Comment (lstein) [inline ldm/invoke/generator/diffusers_pipeline.py:405]: Is this necessary? Under diffusers 0.15 it seems like it is always self.config, and the other changes made in this PR would seem to make the code incompatible with 0.14.

**Review by lstein — APPROVED** [review]
I committed a couple of small fixes, and have a question about the use of `self.config` above. Otherwise looks great and passed a bit of testing.

File: ldm/invoke/generator/diffusers_pipeline.py (hunk 2 above)
Comment (lstein) [reply ldm/invoke/generator/diffusers_pipeline.py:405]: Ok, I tested with and without the conditional for self.config and it failed without the test. So I withdraw this comment.

**Review by ebr — APPROVED** [review]

File: ldm/invoke/generator/diffusers_pipeline.py (hunk 2 above)
Comment (patrickvonplaten) [reply ldm/invoke/generator/diffusers_pipeline.py:405]: Hmm, ideally this shouldn't be necessary anymore. I'll try to open a PR to fix it

File: ldm/models/diffusion/cross_attention_control.py [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
 from compel.cross_attention_control import Arguments
 from diffusers.models.unet_2d_condition import UNet2DConditionModel
-from diffusers.models.cross_attention import AttnProcessor
```
Comment (patrickvonplaten) [inline ldm/models/diffusion/cross_attention_control.py:17]: Did this break? We tried to deprecate the naming so it should have still worked but thrown a warning - maybe we messed up something

Comment (patrickvonplaten) [reply ldm/models/diffusion/cross_attention_control.py:17]: I see this still works, but:
```py
from diffusers.models.attention import CrossAttention
```

stopped working

**Comment by lstein** [comment]
@damian0815 , @blessedcoolant We'll need a codeowner review on this. Thanks!

**Review by blessedcoolant — APPROVED** [review]

**Comment by lstein** [comment]
@damian0815 @blessedcoolant We need a codeowner review on this PR. Thanks!

**Review by blessedcoolant — APPROVED** [review]

**Review by blessedcoolant — APPROVED** [review]

**Review by GreggHelt2 — APPROVED** [review]
Looks good. And thanks for the clear explanation of class renaming etc in the PR message!

**Pull request merged**